src/index.js

```diff
@@ -0,0 +1,163 @@
+import 'lodash';
+
+// SELECT ELEMENTS
+const form = document.getElementById('todoform');
+const todoInput = document.getElementById('newtodo');
+const todosListEl = document.getElementById('todos-list');
+const notificationEl = document.querySelector('.notification');
+
+// VARS
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let EditTodoId = -1;
+
+// RENDER TODOS
+function renderTodos() {
+  if (todos.length === 0) {
+    todosListEl.innerHTML = '<center>Nothing to do!</center>';
+    return;
+  }
+
+  // CLEAR ELEMENT BEFORE A RE-RENDER
+  todosListEl.innerHTML = '';
+
+  // RENDER TODOS
+  todos.forEach((todo, index) => {
+    todosListEl.innerHTML += `
+    <div class="todo" id=${index}>
+      <i 
+        class="bi ${todo.checked ? 'bi-check-circle-fill' : 'bi-circle'}"
+        style="color : ${todo.color}"
+        data-action="check"
+      ></i>
+      <p class="${todo.checked ? 'checked' : ''}" data-action="check">${todo.value}</p>
+      <i class="bi bi-pencil-square" data-action="edit"></i>
+      <i class="bi bi-trash" data-action="delete"></i>
+    </div>
+    `;
+  });
+}
+
+// CHECK A TODO
+function checkTodo(todoId) {
+  todos = todos.map((todo, index) => ({
+    ...todo,
+    checked: index === todoId ? !todo.checked : todo.checked,
+  }));
+
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
+
+// EDIT A TODO
+function editTodo(todoId) {
+  todoInput.value = todos[todoId].value;
+  EditTodoId = todoId;
+}
+
+// DELETE TODO
+function deleteTodo(todoId) {
+  todos = todos.filter((todo, index) => index !== todoId);
+  EditTodoId = -1;
+
+  // re-render
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
+
+// CLICK EVENT LISTENER FOR ALL THE TODOS
+todosListEl.addEventListener('click', (event) => {
+  const { target } = event;
+  const parentElement = target.parentNode;
+
+  if (parentElement.className !== 'todo') return;
+
+  // t o d o id
+  const todo = parentElement;
+  const todoId = Number(todo.id);
+
+  // target action
+  const { action } = target.dataset;
+
+  if (action === 'check') {
+    checkTodo(todoId);
+  } else if (action === 'edit') {
+    editTodo(todoId);
+  } else {
+    deleteTodo(todoId);
+  }
+  // function to delete selected item
+
+  // if (isEmpty) {
+  //   showNotification("Todo's input is empty");
+  // } else if (isDuplicate) {
+  //   showNotification('Todo already exists!');
+  // } else {
+
+  // if (action === 'edit') {
+  //   editTodo(todoId);
+  // }
+
+  // if (action === 'delete') {
+  //   deleteTodo(todoId)
+  // }
+  // action === 'check' && checkTodo(todoId);
+  // action === 'edit' && editTodo(todoId);
+  // action === 'delete' && deleteTodo(todoId);
+});
```

Comment:
- I'd like to kindly suggest that you remove all commented line of code from your code base to make your work look more professional and readable. 🤗